app/components/school-session-type-form.js

```diff
@@ -1,159 +1,102 @@
+import { tracked } from '@glimmer/tracking';
+import { action, computed } from '@ember/object';
 import { inject as service } from '@ember/service';
-import { isPresent } from '@ember/utils';
-import { task } from 'ember-concurrency';
-import ValidationErrorDisplay from 'ilios-common/mixins/validation-error-display';
-import { validator, buildValidations } from 'ember-cp-validations';
-
-const Validations = buildValidations({
-  title: [
-    validator('presence', true),
-    validator('length', {
-      max: 100
-    })
-  ],
-  calendarColor: [
-    validator('presence', true),
-    validator('format', {
-      regex: /^#[a-fA-F0-9]{6}$/,
-    })
-  ],
-  selectedAamcMethodId: [
-    validator(function(aamcMethodId, options, model) {
-      if (!aamcMethodId) {
-        return true;
-      }
-      const assessment = model.get('assessment');
-      const lookup = assessment?'AM':'IM';
-      if (aamcMethodId.indexOf(lookup) !== 0) {
-        return this.createErrorMessage(options.messageKey, aamcMethodId, options);
-      }
-
-      return true;
-    }, {
-      dependentKeys: ['model.assessment'],
-      descriptionKey: 'general.aamcMethod',
-      messageKey: 'general.invalid',
-    })
-  ]
-});
-
-export default Component.extend(ValidationErrorDisplay, Validations, {
-  store: service(),
-
-  classNames: ['school-session-type-form'],
-
-  assessment: false,
-  calendarColor: null,
-  canEditAamcMethod: false,
-  canEditActive: false,
-  canEditAssessment: false,
-  canEditAssessmentOption: false,
-  canEditCalendarColor: false,
-  canEditTitle: false,
-  canUpdate: false,
-  isActive: true,
-  selectedAamcMethodId: null,
-  selectedAssessmentOptionId: null,
-  title: null,
-
-  assessmentOptions: computed(async function() {
-    const store = this.store;
-    return await store.findAll('assessment-option');
-  }),
-
-  allAamcMethods: computed(async function() {
-    const store = this.store;
-    const aamcMethods = await store.findAll('aamc-method');
-
-    return aamcMethods;
-  }),
-
-  filteredAamcMethods: computed('allAamcMethods.[]', 'assessment', 'selectedAamcMethodId', async function() {
-    const assessment = this.assessment;
-    const aamcMethods = await this.allAamcMethods;
-    const selectedAamcMethodId = this.selectedAamcMethodId;
-    return aamcMethods.filter(aamcMethod => {
+import { dropTask, restartableTask } from 'ember-concurrency-decorators';
+import { validatable, IsHexColor, Length, NotBlank } from 'ilios-common/decorators/validation';
+
+@validatable
+export default class SchoolSessionTypeFormComponent extends Component {
+  @service store;
+  @tracked assessment;
+  @tracked @NotBlank() @IsHexColor() calendarColor;
+  @tracked isActive;
+  @tracked selectedAamcMethodId;
+  @tracked selectedAssessmentOptionId;
+  @tracked @NotBlank() @Length(1, 100) title;
+  @tracked assessmentOptions = [];
+  @tracked aamcMethods = [];
+
+  @computed('aamcMethods', 'assessment', 'selectedAamcMethodId')
```

Comment:
If this is needed that seems like an ember bug. Since these values are marked as tracked and accessed synchronously it shouldn't be necessary to call them out manually.

Reply:
you are right - this turns out to be gunk. i had a bit of a hard time to get those getters to work properly on first try, this was one of the things i tried. obviously it got fixed some other way, but then i forgot to remove those annotations.
anyway - removed.